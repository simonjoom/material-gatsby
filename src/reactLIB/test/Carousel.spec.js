import React from 'react';
import { shallow } from 'enzyme';
import Carousel from '../src/Carousel';
import mocker from './helper/mocker';

describe('<Carousel />', () => {
  let wrapper;
  const carouselMock = jest.fn();
  const restore = mocker('carousel', carouselMock);

  afterAll(() => {
    restore();
  });

  const images = [
    'https://lorempixel.com/250/250/nature/1',
    'https://lorempixel.com/250/250/nature/2',
    'https://lorempixel.com/250/250/nature/3',
    'https://lorempixel.com/250/250/nature/4'
  ];

  test('renders', () => {
    wrapper = shallow(<Carousel images={images} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders fixed items', () => {
    const fixedItem = <span>Do you rock!?</span>;
    wrapper = shallow(<Carousel images={images} fixedItem={fixedItem} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('handles content slides', () => {
    const child = (
      <div className="red">
        <h2>First Panel</h2>
        <p>This is your first panel</p>
      </div>
    );
    wrapper = shallow(<Carousel images={images}>{child}</Carousel>);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts className props', () => {
    wrapper = shallow(<Carousel images={images} className="center" />);
    expect(wrapper.hasClass('carousel center')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts external `id` value', () => {
    wrapper = shallow(<Carousel images={images} carouselId="ID" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('handles full width sliders', () => {
    wrapper = shallow(
      <Carousel images={images} options={{ fullWidth: true }} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(carouselMock).toHaveBeenCalledWith({ fullWidth: true });
  });

  test('more options', () => {
    const options = { padding: 12, fullWidth: true, indicators: false };
    wrapper = shallow(<Carousel images={images} options={options} />);
    expect(carouselMock).toHaveBeenCalledWith(options);
  });
});
