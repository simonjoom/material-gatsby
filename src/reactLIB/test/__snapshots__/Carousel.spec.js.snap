// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> accepts className props 1`] = `
<div
  className="carousel center"
>
  <a
    className="carousel-item"
    key=".0"
  >
    <img
      src="https://lorempixel.com/250/250/nature/1"
    />
  </a>
  <a
    className="carousel-item"
    key=".1"
  >
    <img
      src="https://lorempixel.com/250/250/nature/2"
    />
  </a>
  <a
    className="carousel-item"
    key=".2"
  >
    <img
      src="https://lorempixel.com/250/250/nature/3"
    />
  </a>
  <a
    className="carousel-item"
    key=".3"
  >
    <img
      src="https://lorempixel.com/250/250/nature/4"
    />
  </a>
</div>
`;

exports[`<Carousel /> accepts external \`id\` value 1`] = `
<div
  className="carousel"
  id="ID"
>
  <a
    className="carousel-item"
    key=".0"
  >
    <img
      src="https://lorempixel.com/250/250/nature/1"
    />
  </a>
  <a
    className="carousel-item"
    key=".1"
  >
    <img
      src="https://lorempixel.com/250/250/nature/2"
    />
  </a>
  <a
    className="carousel-item"
    key=".2"
  >
    <img
      src="https://lorempixel.com/250/250/nature/3"
    />
  </a>
  <a
    className="carousel-item"
    key=".3"
  >
    <img
      src="https://lorempixel.com/250/250/nature/4"
    />
  </a>
</div>
`;

exports[`<Carousel /> handles content slides 1`] = `
<div
  className="carousel"
>
  <div
    className="red carousel-item"
    key=".0"
  >
    <h2>
      First Panel
    </h2>
    <p>
      This is your first panel
    </p>
  </div>
</div>
`;

exports[`<Carousel /> handles full width sliders 1`] = `
<div
  className="carousel carousel-slider"
>
  <a
    className="carousel-item"
    key=".0"
  >
    <img
      src="https://lorempixel.com/250/250/nature/1"
    />
  </a>
  <a
    className="carousel-item"
    key=".1"
  >
    <img
      src="https://lorempixel.com/250/250/nature/2"
    />
  </a>
  <a
    className="carousel-item"
    key=".2"
  >
    <img
      src="https://lorempixel.com/250/250/nature/3"
    />
  </a>
  <a
    className="carousel-item"
    key=".3"
  >
    <img
      src="https://lorempixel.com/250/250/nature/4"
    />
  </a>
</div>
`;

exports[`<Carousel /> renders 1`] = `
<div
  className="carousel"
>
  <a
    className="carousel-item"
    key=".0"
  >
    <img
      src="https://lorempixel.com/250/250/nature/1"
    />
  </a>
  <a
    className="carousel-item"
    key=".1"
  >
    <img
      src="https://lorempixel.com/250/250/nature/2"
    />
  </a>
  <a
    className="carousel-item"
    key=".2"
  >
    <img
      src="https://lorempixel.com/250/250/nature/3"
    />
  </a>
  <a
    className="carousel-item"
    key=".3"
  >
    <img
      src="https://lorempixel.com/250/250/nature/4"
    />
  </a>
</div>
`;

exports[`<Carousel /> renders fixed items 1`] = `
<div
  className="carousel"
>
  <div
    className="carousel-fixed-item center"
  >
    <span>
      Do you rock!?
    </span>
  </div>
  <a
    className="carousel-item"
    key=".0"
  >
    <img
      src="https://lorempixel.com/250/250/nature/1"
    />
  </a>
  <a
    className="carousel-item"
    key=".1"
  >
    <img
      src="https://lorempixel.com/250/250/nature/2"
    />
  </a>
  <a
    className="carousel-item"
    key=".2"
  >
    <img
      src="https://lorempixel.com/250/250/nature/3"
    />
  </a>
  <a
    className="carousel-item"
    key=".3"
  >
    <img
      src="https://lorempixel.com/250/250/nature/4"
    />
  </a>
</div>
`;
